libs/ui/src/components/activity-section-list/activity-section-list.tsx

```diff
@@ -22,26 +34,38 @@ interface Props {
   tokenAddress?: string;
 }
 
-const keyExtractor = ({ hash }: ActivityData) => hash;
+const keyExtractor = ({ hash, timestamp }: ActivityData) => `${hash}_${timestamp}`;
+
+const renderSectionHeader = (item: { section: SectionListData<ActivityData, SectionListActivityData> }) => (
+  <Row style={styles.dateWrapper}>
+    <Text style={styles.dateText}>{item.section.title}</Text>
+  </Row>
+);
 
 const emptyIconSize = getCustomSize(isWeb ? 30 : 36);
 
 export const ActivitySectionList: FC<Props> = ({ publicKeyHash, chainId, tokenAddress = '' }) => {
+  const [offsetY, setOffsetY] = useState(0);
   const { activity, fetchData, isLoading } = useAllActivity(publicKeyHash, getDebankId(chainId), tokenAddress);
 
-  useEffect(() => {
+  useTimerEffect(() => fetchData(0), DATA_UPDATE_TIME, [publicKeyHash, chainId, offsetY], offsetY);
+
+  const handleScroll = debounce((event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    setOffsetY(event.nativeEvent.contentOffset.y);
+  }, DEBOUNCE_TIME);
+
+  const handleEndReached = () => {
     fetchData();
-  }, []);
+  };
 
-  const renderItem: SectionListRenderItem<ActivityData, SectionListActivityData> = ({ item: activityItems }) => (
-    <ActivityList transaction={activityItems} address={publicKeyHash} chainName={getDebankId(chainId)} />
+  const renderItem: SectionListRenderItem<ActivityData, SectionListActivityData> = useCallback(
+    ({ item: activityItems }) => (
+      <ActivityList transaction={activityItems} address={publicKeyHash} chainName={getDebankId(chainId)} />
+    ),
+    [publicKeyHash, chainId]
   );
 
-  const renderSectionHeader = (item: { section: SectionListData<ActivityData, SectionListActivityData> }) => (
-    <Row style={styles.dateWrapper}>
-      <Text style={styles.dateText}>{item.section.title}</Text>
-    </Row>
-  );
+  const getItemLayout = getItemLayoutSectionList<ActivityData, SectionListActivityData>(getCustomSize(9));
```

Comment:
on each render new `getItemLayout` will be created. move it out of the component